torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1351,11 +1353,80 @@ def call_function(
             body_r.as_proxy(),
         )
 
+        if self.value.__name__ == "hints_wrapper":
+            if "hints" not in kwargs:
+                unimplemented("key hints not found in kwargs")
+            # make hints into p_kwargs
+            p_kwargs["hints"] = kwargs["hints"].as_python_constant()
         return _call_function_and_unflatten_output(
             tx, self.value, tuple(p_args), p_kwargs, flat_example_value, treespec
         )
 
 
+class HintsWrapperHigherOrderVariable(TorchHigherOrderOperatorVariable):
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        _check_supported_callable_arg(tx, args[0], "body_fn")
+
+        # inputs
+        if not isinstance(args[1], (ListVariable, TupleVariable)):
+            unimplemented(
+                f"Expected a tuple but got {args[1].python_type()}",
+            )
+        operands = args[1].unpack_var_sequence(tx)
+
+        if not isinstance(args[2], ConstDictVariable):
+            unimplemented(
+                f"Expected a dict but got {args[2].python_type()}",
+            )
+
+        (
+            (body_r, treespec),
+            body_graph,
+            body_lifted_freevars,
+        ) = speculate_subgraph(
+            tx,
+            args[0],  # function
+            operands,
+            args[2].as_python_constant(),
+            "hints_wrapper",
+            source_target=self.value,
+            should_flatten_outputs=True,
+        )
+
+        body_gmod = torch.fx.GraphModule(tx.output.nn_modules, body_graph)
+        body_name = add_subgraph(
+            tx,
+            "hints_wrapper_body",
+            body_gmod,
+        )
+
+        body_node = make_attr(tx, body_name)
+
+
+        # Since, we call `speculate_subgraph` with `set_subgraph_inputs="automatic`,
+        # all the arguments are lifted.
+        lifted_args = tuple(arg for arg in body_lifted_freevars.keys())
+        p_args = (body_node, lifted_args, {})
+
+        p_kwargs = {}
+        if "hints" not in kwargs:
+            raise Unsupported("hints_wrapper - key hints not provided")
```

Comment:
Can move it to input checking section (to skip speculate_subgraph when hints don't exist)?  Change to IncorrectUsage error? Might also add a unit test that self.assertRaisesRegex the error to guard any change.

Reply:
Sounds good. I will change.